src/picosvg/svg_types.py

```diff
@@ -286,14 +286,13 @@ def callback(curr_xy, cmd, args, prev_xy, prev_cmd, prev_args)
                 if y_coord_idxs:
                     new_y += new_cmd_args[y_coord_idxs[-1]]
 
-                # z takes the pen to the return_pos
                 if new_cmd.lower() == "z":
-                    new_x, new_y = return_pos
+                    new_x, new_y = subpath_start_pos
 
-                prev_pos = copy.copy(curr_pos)
+                prev_pos = curr_pos
                 curr_pos = Point(new_x, new_y)
```

Comment:
you could even swap in one line if you like `prev_pos, curr_pos = curr_pos, Point(new_x, new_y)`

Reply:
I always forget that works, I like it!

Reply:
I can even do new_x, new_y = curr_pos above! Amazing :D